src/Helsenorge.Messaging/Security/SignThenEncryptMessageProtection.cs

```diff
@@ -261,5 +266,20 @@ private byte[] Unprotect(byte[] data, X509Certificate2 signingCertificate)
             // Return the raw content (without a signature).
             return signedCms.ContentInfo.Content;
         }
+        
+        private EnvelopedCms GetEnvelope(byte[] rawContent)
+        {
+            if (_messagingEncryptionType == MessagingEncryptionType.AES256)
+            {
+                return new EnvelopedCms(new ContentInfo(rawContent));
+            }
+
+            else if (_messagingEncryptionType == MessagingEncryptionType.TripleDES)
+            {
+                return new EnvelopedCms(new ContentInfo(rawContent), new AlgorithmIdentifier(new Oid("1.2.840.113549.3.7")));
+            }
+
+            return null;
```

Comment:
A simpler and more fail-safe approach to this would be:

We always want to default to AES-256
```
private EnvelopedCms GetEnvelope(byte[] rawContent)
{
    if (_messagingEncryptionType == MessagingEncryptionType.TripleDES)
        return new EnvelopedCms(new ContentInfo(rawContent), new AlgorithmIdentifier(new Oid("1.2.840.113549.3.7")));
    
    return new EnvelopedCms(new ContentInfo(rawContent));
}
```